packages/jetpack-redirects/src/index.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Builds an URL using the jetpack.com/redirect service
+ *
+ * If $source is a simple slug, it will be sent using the source query parameter. e.g. jetpack.com/redirect?source=slug
+ *
+ * If $source is a full URL, starting with https://, it will be sent using the url query parameter. e.g. jetpack.com/redirect?url=https://wordpress.com
```

Comment:
It's weird that the `source` parameter would have multiple meanings like this. Especially if the `getRedirectUrl` function supports named arguments in `args`. What if `getRedirectUrl` had only one parameter, `args`, and `source` and `url` were among the supported keys?

Reply:
I agree but we have a backward compatibility concern there.

This package will replace this lib https://github.com/Automattic/jetpack/blob/master/_inc/client/lib/jp-redirect/index.js

which is used throughout the code.

The possibility to use an URL instead of a source slug was added later, and we the function is already used in hundreds of places and we didn't want to have to change them all...

Reply:
Alright, let's keep the API as it is :+1: